AsyncNats/Messages/NatsUnsub.cs

```diff
@@ -21,10 +22,12 @@ public async ValueTask Serialize(PipeWriter writer)
             await writer.WriteAsync(Encoding.UTF8.GetBytes($"UNSUB {SubscriptionId} {MaxMessages}\r\n"));
         }
 
-        public static IMemoryOwner<byte> RentedSerialize(NatsMemoryPool pool, string subscriptionId, int? maxMessages)
+        public static IMemoryOwner<byte> RentedSerialize(NatsMemoryPool pool, long subscriptionId, int? maxMessages)
         {
+            byte[] subscriptionBytes = Encoding.UTF8.GetBytes(subscriptionId.ToString());
```

Comment:
Using Utf8Formatter will probably be faster (you can skip 2 alloc's):
Span<byte> subscriptionBytes = stackalloc byte[32]; // I think max ?20?
Utf8Formatter.TryFormat(subscriptionId, subscriptionBytes, out var subscriptionLength); // No need to check return, buffer is enough to hold largest int64 [but feel free to throw exception on 'false'

Reply:
Yes, good one!

Reply:
True, but if we're going for the max, lets go for the max ;)